itest/lnd_open_channel_test.go

```diff
@@ -822,3 +822,53 @@ func testSimpleTaprootChannelActivation(ht *lntest.HarnessTest) {
 	// Our test is done and Alice closes her channel to Bob.
 	ht.CloseChannel(alice, chanPoint)
 }
+
+// testOpenChannelLockedBalance tests that when a funding reservation is
+// made for opening a channel, the balance of the required outputs shows
+// up as locked balance in the WalletBalance response.
+func testOpenChannelLockedBalance(ht *lntest.HarnessTest) {
+	// Before we start the test, we'll ensure both sides are connected so
+	// the funding flow can be properly executed.
+	alice := ht.Alice
+	bob := ht.Bob
+	ht.EnsureConnected(alice, bob)
+
+	// We first make sure Alice has no locked wallet balance.
+	balance := alice.RPC.WalletBalance()
+	require.Equal(ht, int64(0), balance.LockedBalance)
+
+	// Next, we register a ChannelAcceptor on Bob. This way, we can get
+	// Alice's wallet balance after coin selection is done and outpoints
+	// are locked.
+	stream, cancel := bob.RPC.ChannelAcceptor()
+	defer cancel()
+
+	// Then, we request creation of a channel from Alice to Bob. We don't
+	// do it synchronously since we want to receive Bob's message in the
+	// same goroutine.
+	openChannelReq := &lnrpc.OpenChannelRequest{
+		NodePubkey:         bob.PubKey[:],
+		LocalFundingAmount: int64(funding.MaxBtcFundingAmount),
+	}
+	_ = alice.RPC.OpenChannel(openChannelReq)
+
+	// After that, we receive the request on Bob's side, get the wallet
+	// balance from Alice, and ensure it's non-zero.
+	req, err := stream.Recv()
+	require.NoError(ht, err)
+
+	balance = alice.RPC.WalletBalance()
+	require.NotEqual(ht, int64(0), balance.LockedBalance)
```

Comment:
This is where the itest fails before the fix. `LockedBalance` only counts outputs that are locked via `LeaseOutput`, leaving out ones that are locked via `LockOutpoint`. This results in 0 locked balance, though funds are reserved for opening the channel.